### PR TITLE
Mypy check2

### PR DESCRIPTION
**Pull request by palango**
Needs to be rebased once #2814 is merged

This are only typing improvements.

Part of #2797

### CODE REVIEW

File: raiden/transfer/channel.py [hunk 1]
```diff
@@ -850,8 +856,8 @@ def get_current_balanceproof(end_state: NettingChannelEndState) -> BalanceProofD
     else:
         locksroot = EMPTY_MERKLE_ROOT
         nonce = 0
-        transferred_amount = 0
-        locked_amount = 0
+        transferred_amount: typing.TokenAmount = 0
+        locked_amount: typing.Balance = 0
```
Comment (hackaugusto) [inline raiden/transfer/channel.py]: Humm, I think we should use `TokenAmount` for this one

File: raiden/transfer/node.py [hunk 2]
```diff
@@ -907,7 +907,11 @@ def handle_state_change(chain_state: ChainState, state_change: StateChange) -> T
     return iteration
 
 
-def is_transaction_effect_satisfied(chain_state, transaction, state_change) -> bool:
+def is_transaction_effect_satisfied(
+        chain_state: ChainState,
+        transaction,
```
Comment (hackaugusto) [inline raiden/transfer/node.py]: what about this one :) ? These are all instances of `ContractSendEvent`

File: raiden/transfer/mediated_transfer/initiator.py [hunk 3]
```diff
@@ -74,23 +74,27 @@ def handle_block(
...
-        block_number=state_change.block_number,
-        lock_expiration_threshold=lock_expiration_threshold,
-    )
-    if lock_has_expired:
-        expired_lock_events = channel.events_for_expired_lock(
-            channel_state,
-            secrethash,
-            locked_lock,
-            pseudo_random_generator,
+    if locked_lock:
```
Comment (LefterisJP) [inline raiden/transfer/mediated_transfer/initiator.py]: Is this the bug you meant this is fixing?  So if the secrethash is not in the secrethashes to locked locks mapping?

Note: I would do:

```
if not locked_lock:
    return TransitionResult(initiator_state, list())

...
...
```

This way you don't need to have an extra level of indentation.

Comment (palango) [reply raiden/transfer/mediated_transfer/initiator.py]: Yes, and this code also had problems `locked_lock and channel.is_lock_expired(`

**Comment by palango** [comment]
@LefterisJP Fixed your remark and rebased.

**Pull request merged**